src/js/components/DescriptionList.js

```diff
@@ -23,11 +23,14 @@ class DescriptionList extends React.Component {
 
       // Check whether we are trying to render an object that is not a
       // React component
-      if (typeof value === 'object' && !Array.isArray(value) &&
-        value !== null && !React.isValidElement(value)) {
+      if (typeof value === 'object' && value !== null &&
```

Comment:
We could join the index and the parent key/heading, but I will not iterate on this until we found a visually appealing way as the current styling neither works for "normal" object sublist nor array based sublist. So this change might get dropped.